test/suites/ipv6/dualstack.robot

```diff
@@ -0,0 +1,178 @@
+
+
+*** Keywords ***
+Setup
+    [Documentation]    Test suite setup
+    Initialize Global Variables
+    Login MicroShift Host
+    Setup Suite With Namespace
+    Wait Until Greenboot Health Check Exited
+
+Teardown
+    [Documentation]    Test suite teardown
+    Teardown Suite With Namespace
+    Logout MicroShift Host
+
+Initialize Global Variables
+    [Documentation]    Initializes global variables.
+    Log    IP1: ${USHIFT_HOST_IP1} IPv6: ${USHIFT_HOST_IP2}
+    Should Not Be Empty    ${USHIFT_HOST_IP1}    USHIFT_HOST_IP1 variable is required
+    Should Not Be Empty    ${USHIFT_HOST_IP2}    USHIFT_HOST_IP2 variable is required
+
+Migrate To Dual Stack
+    [Documentation]    Configure MicroShift to enable dual stack network
+
+    ${dual_stack}=    CATENATE    SEPARATOR=\n
+    ...    ---
+    ...    network:
+    ...    \ \ clusterNetwork: [10.42.0.0/16, fd01::/48]
+    ...    \ \ serviceNetwork: [10.43.0.0/16, fd02::/112]
+    ${replaced}=    Replace MicroShift Config    ${dual_stack}
+    Upload MicroShift Config    ${replaced}
+    Restart MicroShift
+
+Delete Hello MicroShift Route
+    [Documentation]    Delete route for cleanup.
+    Oc Delete    route/hello-microshift -n ${NAMESPACE}
+
+Wait For Service Deletion With Timeout
+    [Documentation]    Polls for service and endpoint by "app=hello-microshift" label. Fails if timeout
+    ...    expires. This check is unique to this test suite because each test here reuses the same namespace. Since
+    ...    the tests reuse the service name, a small race window exists between the teardown of one test and the setup
+    ...    of the next. This produces flakey failures when the service or endpoint names collide.
+    Wait Until Keyword Succeeds    30s    1s
+    ...    Network APIs With Test Label Are Gone
+
+Expose Hello MicroShift Service Via Route IPv4
+    [Documentation]    Expose the "hello microshift" application through the Route
+    Run With Kubeconfig    oc apply -n ${NAMESPACE} -f assets/hello-microshift-service.yaml
+    Oc Expose    svc hello-microshift --hostname ${HOSTNAME} -n ${NAMESPACE}
+
+Expose Hello MicroShift Service Via Route IPv6
+    [Documentation]    Expose the "hello microshift" application through the Route
+    Run With Kubeconfig    oc apply -n ${NAMESPACE} -f assets/hello-microshift-service-ipv6.yaml
+    Oc Expose    svc hello-microshift --hostname ${HOSTNAME} -n ${NAMESPACE}
+
+Network APIs With Test Label Are Gone
+    [Documentation]    Check for service and endpoint by "app=hello-microshift" label. Succeeds if response matches
+    ...    "No resources found in <namespace> namespace." Fail if not.
+    ${match_string}=    Catenate    No resources found in    ${NAMESPACE}    namespace.
+    ${match_string}=    Remove String    ${match_string}    "
+    ${response}=    Run With Kubeconfig    oc get svc,ep -l app\=hello-microshift -n ${NAMESPACE}
+    Should Be Equal As Strings    ${match_string}    ${response}    strip_spaces=True
```

Comment:
Including this place, this code makes 3 appearances - shall we move it to resources/ ?

Reply:
You are right, but I would rather do this in a different change as it touches non-ipv6 tests too?